src/app/client/src/app/modules/shared/components/sb-datatable/sb-datatable.component.ts

```diff
@@ -0,0 +1,85 @@
+import { filter } from 'rxjs/operators';
+import { Component, OnInit, Input, OnChanges } from '@angular/core';
+import { ExportToCsv } from 'export-to-csv';
+import * as _ from 'lodash-es';
+
+@Component({
+  selector: 'sb-datatable',
+  templateUrl: './sb-datatable.component.html',
+  styleUrls: ['./sb-datatable.component.scss']
+})
+export class SbDatatableComponent implements OnInit, OnChanges {
+
+  @Input() searchFields;
+  @Input() data;
+  @Input() columns;
+  @Input() downloadCSV;
+  @Input() sortable;
+  public tableData = [];
+  public searchData;
+  public sortOrder = 'asc';
+  public sortField = 'state';
+  public showLoader = false;
+  public csvExporter: any;
+
+  constructor() { }
+
+  ngOnInit() {
+  }
+
+  ngOnChanges() {
+    this.tableData = _.cloneDeep(this.data);
+  }
+  search() {
+    if (this.searchFields && this.searchFields.length !== 0) {
+      if (this.searchData) {
+        this.tableData = this.filterData(this.data);
+      } else {
+        this.tableData = this.data
+      }
+    } else {
+      this.tableData = _.filter(this.data, (row) => {
+        return (_.lowerCase(JSON.stringify(row))).includes(_.lowerCase(this.searchData))
+      })
+    }
+
+  }
+  filterData(data) {
+    let resultData = [];
+    _.forEach(data, (row, index) => {
```

Comment:
User array or library filter method or create the angular pipe to filter.
Dont use, nested forEach loop. This will cause performance issue.